third-party/realdds/src/dds-participant.cpp

```diff
@@ -105,7 +105,10 @@ struct dds_participant::listener_impl : public eprosima::fastdds::dds::DomainPar
                 std::lock_guard< std::mutex > lock( participants_mutex );
                 participant_info pinfo( info.info.m_participantName, info.info.m_guid.guidPrefix );
                 name = pinfo.name;
-                LOG_DEBUG( _owner.name() << ": +participant '" << name << "' " << guid );
+                // When remote participant is active before this participant, listener callback can happen before
+                // create_participant funciton returns and _owner.name() can be used.
+                rsutils::string::slice & owner_name = _owner.get() ? _owner.name() : rsutils::string::slice( "" );
```

Comment:
I don't think you should take a reference to a temporary object. Better would not a value.
If creating a slice for nothing, pass in a length of 0 rather than calling strlen.
I'd prefer:
```
rsutils::string::slice owner_name;;
if( _owner.get() )
    owner_name = _owner.name();
```